lang/rust/avro/src/schema.rs

```diff
@@ -2365,4 +2513,560 @@ mod tests {
             _ => panic!("Expected an Error::InvalidSchemaName!"),
         }
     }
+
+    #[test]
+    fn avro_3448_test_proper_resolution_inner_record_inherited_namespace() {
+        let schema = r#"
+        {
+          "name": "record_name",
+          "namespace": "space",
+          "type": "record",
+          "fields": [
+            {
+              "name": "outer_field_1",
+              "type": [
+                        "null",
+                        {
+                            "type":"record",
+                            "name":"inner_record_name",
+                            "fields":[
+                                {
+                                    "name":"inner_field_1",
+                                    "type":"double"
+                                }
+                            ]
+                        }
+                    ]
+            },
+            {
+                "name": "outer_field_2",
+                "type" : "inner_record_name"
+            }
+          ]
+        }
+        "#;
+        let schema = Schema::parse_str(schema).unwrap();
+        let rs = ResolvedSchema::from(&schema);
```

Comment:
I think it would be good to assert here that `rs.get_names()` has size = 2.
Same for the following tests.